spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -872,11 +876,13 @@ private void invokeListener(final ConsumerRecords<K, V> records) {
 
 		private void invokeBatchListener(final ConsumerRecords<K, V> records) {
 			List<ConsumerRecord<K, V>> recordList = new LinkedList<ConsumerRecord<K, V>>();
```

Comment:
May we consider do not initialize this variable if we are not interested in the list, but the whole `records` ?